src/mito2/src/region/opener.rs

```diff
@@ -249,12 +249,28 @@ impl RegionOpener {
     }
 
     /// Returns a new manifest options.
-    fn manifest_options(&self, config: &MitoConfig) -> RegionManifestOptions {
-        RegionManifestOptions {
+    fn manifest_options(&self, config: &MitoConfig) -> Result<RegionManifestOptions> {
+        let options = RegionOptions::try_from(&self.options)?;
+        let object_store = self.object_store(&options.storage)?.clone();
+        Ok(RegionManifestOptions {
             manifest_dir: new_manifest_dir(&self.region_dir),
-            object_store: self.object_store.clone(),
+            object_store,
             compress_type: config.manifest_compress_type,
             checkpoint_distance: config.manifest_checkpoint_distance,
+        })
+    }
+
+    /// Returns an object store corresponding to the name. If the object sotre doesn't exist, this method returns the default object store.
```

Comment:
This method returns an error if the object store doesn't exist, so I guess you want to document that it returns the default one if `name` is `None`.
```suggestion
    /// Returns an object store corresponding to the name. If `name` is `None`, this method returns the default object store.
```